### DISCUSSION TITLE
Get state during prerender?

### DISCUSSION THREAD

**Discussion by clsferguson** [mrpmorris/Fluxor#464] [Q&A]
Is there a way to get the state of something during the prerender, while using serverprerender? So it can be loaded and available for the final render?

Right now I have my initial state, but it loads as an empty string. Then an effect is used to get the info from the entity framework database. This data doesn't show up until the prerender is done, causing ui flicker.

I'd like to make it more seamless.

Thanks.

**Comment by janusqa** [comment]
A compromise could be to disable pre-render for that one page.

**Comment by mrpmorris** [comment]
I need to implement IDispatcher.DispatchAsync so Blazor knows the initial dispatch hasn't completed and waits for it before returning the rendered page.
